ui/app/services/token.js

```diff
@@ -42,6 +42,14 @@ export default class TokenService extends Service {
   })
   fetchSelfToken;
 
+  @task(function*(oneTimeToken) {
+    const TokenAdapter = getOwner(this).lookup('adapter:token');
+
+    const token = yield TokenAdapter.exchangeOneTimeToken(oneTimeToken);
+    this.secret = token.secret;
+  })
+  exchangeOneTimeToken;
```

Comment:
⚠️ totally random thoughts that you don't have to take seriously

Hm, even though it's consistent to the file to use an EC task here, looking at this with fresh eyes it seems like a simple promise would suffice? We aren't doing any fancy chaining or error control flow. Idk.

Reply:
I took it seriously in a6e6b94, it makes sense to remove the needless wrapping, thank you!